airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py

```diff
@@ -78,9 +78,12 @@ def failure(response: FacebookResponse, request: Optional[FacebookRequest] = Non
             # although it is Optional in the signature for compatibility, we need it always
             assert request, "Missing a request object"
             resp_body = response.json()
-            if not isinstance(resp_body, dict) or resp_body.get("error", {}).get("code") != FACEBOOK_BATCH_ERROR_CODE:
+            if not isinstance(resp_body, dict) or (resp_body.get("error", {}).get("code") != FACEBOOK_BATCH_ERROR_CODE and resp_body.get("error", {}).get("message") != "Please reduce the amount of data you're asking for, then retry your request"):
```

Comment:
we're doing a similar check twice and the flow is difficult to follow

can we rewrite as 
1. If resp_body is not a dict: raise RuntimeError
2. elif error message == please reduce ...: update the batch size
3. elif error code != FACEBOOK_BATCH_ERROR: raise RuntimeError

two branches will lead to the same RuntimeError, but the conditions will be easier to read